packages/ember-glimmer/tests/integration/helpers/yield-test.js

```diff
@@ -0,0 +1,256 @@
+
+    this.runTask(() => set(this.context, 'boundText', 'outer'));
+    let element2 = this.firstChild;
+
+    this.assertSameNode(element1, element2);
+  }
+
+  ['@test inner keyword doesn\'t mask yield property']() {
+    let KiwiCompComponent = Component.extend({
+      boundText: 'inner'
+    });
+
+    this.registerComponent('kiwi-comp', { ComponentClass: KiwiCompComponent, template: '{{#with boundText as |item|}}<p>{{item}}</p><p>{{yield}}</p>{{/with}}' });
+
+    this.render('{{#kiwi-comp}}{{item}}{{/kiwi-comp}}', { item: 'outer' });
+
+    equal(this.$('div p:contains(inner) + p:contains(outer)').length, 1, 'outer property isn\'t masked by inner keyword on render');
+
+    let element1 = this.firstChild;
+
+    this.assertStableRerender();
+
+    this.runTask(() => set(this.context, 'item', 'otherworld'));
+
+    equal(this.$('div p:contains(inner) + p:contains(otherworld)').length, 1, 'inner component property isn\'t masked by outer keyword when rerendered with new context');
+
+    this.runTask(() => set(this.context, 'item', 'outer'));
+
+    let element2 = this.firstChild;
+
+    this.assertSameNode(element1, element2);
+  }
+
+  ['@test can bind a keyword to a component and use it in yield']() {
+    let KiwiCompComponent = Component.extend({
+      boundText: 'inner'
+    });
+
+    this.registerComponent('kiwi-comp', { ComponentClass: KiwiCompComponent, template: '<p>{{content}}</p><p>{{yield}}</p>' });
+
+    this.render('{{#with boundText as |item|}}{{#kiwi-comp content=item}}{{item}}{{/kiwi-comp}}{{/with}}', { boundText: 'outer' });
+
+    equal(this.$('div p:contains(outer) + p:contains(outer)').length, 1, 'component and yield have keyword');
+
+    let element1 = this.firstChild;
+
+    this.assertStableRerender();
+
+    this.runTask(() => set(this.context, 'boundText', 'update'));
+
+    equal(this.$('div p:contains(update) + p:contains(update)').length, 1, 'keyword has correctly propagated update');
+
+    this.runTask(() => set(this.context, 'boundText', 'outer'));
+
+    let element2 = this.firstChild;
+
+    this.assertSameNode(element1, element2);
+  }
+
+  ['@htmlbars yield view should be a virtual view']() {
+    let ParentCompComponent = Component.extend({
+      isParentComponent: true
+    });
+
+    let ChildCompComponent = Component.extend({
+      didReceiveAttrs() {
+        this._super();
+        let parentView = this.get('parentView');
+
+        ok(parentView.get('isParentComponent'), 'parent view is the parent component');
+      }
+    });
+
+    this.registerComponent('parent-comp', { ComponentClass: ParentCompComponent, template: '{{yield}}' });
+    this.registerComponent('child-comp', { ComponentClass: ChildCompComponent, template: 'dave' });
+
+    this.render('{{#parent-comp}}{{child-comp}}{{/parent-comp}}');
+  }
+
+  ['@test yield should work for views even if parentView is null']() {
+    this.registerComponent('kiwi-comp', { template: 'Layout: {{yield}}' });
```

Comment:
@wycats I can go through all the tests and change the title of the tests to reflect the change to components